src/tests/stats/test_distribution_uniform.fypp

```diff
@@ -0,0 +1,298 @@
+        print *, "Test uniform random generator with chi-squared"
+        freq = 0
+        do i = 1, num
+            j = array_size * uni_rvs( )
+            freq(j) = freq(j) + 1
+        end do
+        chisq = 0.0_dp
+        expct = num / array_size
+        do i = 0, array_size - 1
+           chisq = chisq + (freq(i) - expct) ** 2 / expct
+        end do
+        write(*,*) "The critical value for chi-squared with 1000 d. of f. is"  &
+            //" 1143.92"
+        write(*,*) "Chi-squared for uniform random generator is : ", chisq
+        call check((chisq < 1143.9) ,                                          &
+            msg = "uniform randomness failed chi-squared test", warn=warn)
+    end subroutine test_uni_rvs_0
+
+
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+      subroutine test_uni_rvs_${t1[0]}$${k1}$
+          ${t1}$ :: res(15), scale, loc
+          integer :: i, n, seed, get, k
+
+          #:if k1 == "int8"
+          ${t1}$, parameter :: ans(15) = [47, 99, 43, 37, 48, 30, 27, 100, 30, &
+                                          33, 21, 103, 55, 54, 110]
+          #:elif k1 == "int16"
+          ${t1}$, parameter :: ans(15) = [25, 4, 81, 98, 49, 34, 32, 62, 115,  &
+                                          112, 26, 20, 37, 100, 82]
+          #:elif k1 == "int32"
+          ${t1}$, parameter :: ans(15) = [19, 52, 56, 20, 59, 44, 34, 102, 19, &
+                                          39, 60, 50, 97, 56, 67]
+          #:elif k1 == "int64"
+          ${t1}$, parameter :: ans(15) = [76, 45, 43, 75, 76, 15, 25, 24, 114, &
+                                          113, 94, 29, 109, 93, 89]
+          #:elif t1[0] == "r"
+          ${t1}$, parameter :: ans(15) =                                       &
+                          [0.914826628538749186958511927514337003_${k1}$,      &
+                          0.367330098664966409049981166390352882_${k1}$,       &
+                          1.77591243057709280428468900936422870_${k1}$,        &
+                          0.885921308987590139238932351872790605_${k1}$,       &
+                          0.950735656542987861428173346212133765_${k1}$,       &
+                          -0.659562573857055134407545438079978339_${k1}$,      &
+                          -0.116661718506947176265953203255776316_${k1}$,      &
+                          0.837391893893859151631886561517603695_${k1}$,       &
+                          -0.703954396598600540269075054311542772_${k1}$,      &
+                          0.382592729851141566399519433616660535_${k1}$,       &
+                          -0.132472493978185168472805344208609313_${k1}$,      &
+                          -0.878723366294216184924081858298450243_${k1}$,      &
+                          -0.901660046141515819639877804547722917_${k1}$,      &
+                          -0.164090614147737401395943379611708224_${k1}$,      &
+                          -0.333886718190384290672056977200554684_${k1}$]
+          #:else
```

Comment:
Here I suggest matching `complex` like in the previous cases, just to make the intention extra clear.